src/layouts/Base.astro

```diff
@@ -45,15 +45,16 @@ const socials: Social[] | undefined = siteConfig.socials?.map((social) => {
     name: social.name || "",
     iconImageID: social.icon?.asset?._id || "",
     link: social.link
-      ? getHrefFromLinkObject(social.link, language, preview)
+      ? getHrefFromLinkObject(social.link, getLocale(), preview)
       : "",
-    alt: social.icon?.asset?.altTexts?.[language] || "",
+    alt: social.icon?.asset?.altTexts?.[getLocale()] || "",
   };
 });
 ---
 
-<html class="overflow-x-clip" lang={language.toLowerCase()}>
+<html class="overflow-x-clip" lang={getLocale().toLowerCase()}>
   <head>
+    <I18nClient />
```

Comment:
Also worth checking into is sitemap generation with i18n: https://astro-i18n.netlify.app/recipes/sitemap/ 

I'll also make a small ticket for this in the backlog. Unsure how it works with the astro sitemap we use now